source/SonicHeroes.WidescreenFix/dllmain.cpp

```diff
@@ -121,8 +116,45 @@ void Init()
                 fmul    dword ptr[eax08]
             }
         }
-    }; injector::MakeInline<CutOffAreaHook>(pattern.count(1).get(0).get<uint32_t>(0), pattern.count(1).get(0).get<uint32_t>(17));
-
+    }; injector::MakeInline<CutOffAreaHORHook>(pattern.count(1).get(0).get<uint32_t>(0), pattern.count(1).get(0).get<uint32_t>(17));
+    
+    struct CutOffAreaVertHook
+	{
+		void operator()(injector::reg_pack& regs)
+		{
+            float edx6C = *(float*)(regs.edx + 0x6C) / Screen.fHudZoomScale;
+            float eax10 = *(float*)(regs.eax + 0x10);
+            _asm
+            {
+                fld     dword ptr[edx6C]
+                fmul    dword ptr[eax10]
+            }
+         }
+    }; injector::MakeInline<CutOffAreaVertHook>(pattern.count(1).get(0).get<uint32_t>(42), pattern.count(1).get(0).get<uint32_t>(6));
+    
+    pattern = hook::pattern("83 C3 0C 53 56 E8 ?? ?? ?? ?? ??"); //0x6B7272
```

Comment:
the remaining wildcards are useless and should be removed, besides, there's no point to use '??' instead of '?'.

Reply:
Those wildcards are there incase it finds it in other builds of this game like the trial for instance,

Reply:
They do absolutely nothing currently. Removing them will not change the behavior.